libs/frontend/ui-build/src/components/BuildDndOverlay.vue

```diff
@@ -79,35 +76,150 @@ const overlayStyle = computed(() => {
           overlayHeight = outlineThickness
           overlayBackgroundColor = outlineColor
         } else if (hoverBottom) {
-          overlayTop = elementRelativeBottom - outlineThickness
+          overlayTop = relativeBottom - outlineThickness
           overlayWidth = elementWidth
           overlayHeight = outlineThickness
           overlayBackgroundColor = outlineColor
         }
       }
 
-      // Divide offsets by `builderScale` because `.build-content-window-inner` scales
-      // based on `builderScale` in the editor context.
-      overlayTop /= builderScale
-      overlayLeft /= builderScale
-      overlayWidth /= builderScale
-      overlayHeight /= builderScale
-
-      overlayTop += buildContentWindowInner.scrollTop
-      overlayLeft += buildContentWindowInner.scrollLeft
+      const { top, left, width, height } = normalizeDimensions(
+        overlayTop,
+        overlayLeft,
+        overlayWidth,
+        overlayHeight,
+        buildContentWindowInner,
+      )
 
       return {
-        top: `${overlayTop}px`,
-        left: `${overlayLeft}px`,
-        width: `${overlayWidth}px`,
-        height: `${overlayHeight}px`,
+        top: `${top}px`,
+        left: `${left}px`,
+        width: `${width}px`,
+        height: `${height}px`,
         backgroundColor: overlayBackgroundColor,
         border: overlayBorder,
       }
     }
   }
   return undefined
 })
+
+// Use `ref`+`watch` instead of `computed` because we have to use `nextTick`
+// to wait for the screen to (re)render before calculating the width and
+// height of the component on page load & after selection.
+const selectionOverlayStyle = ref()
+
+watch(
+  () => ({
+    selectedComponent: editor.value?.selectedComponent,
+    // Re-compute selection overlay width & height when active breakpoint changes.
+    activeBreakpoint: activeBreakpoint.value,
+  }),
+  async ({ selectedComponent }) => {
+    if (selectedComponent) {
+      // This `nextTick` is necessary for buildContentWindowInner and componentElement
+      await nextTick()
```

Comment:
Here the style for selection overlay is calculated with `ref`+`watch` instead of `computed`, because I noticed the width&height for the bounding rect of the selected component is sometimes incorrect when a component selected. The dimensions is also incorrect when a component is removed & recovered. For example:

https://github.com/samatechtw/pubstudio-builder/assets/59676941/d1545b75-d059-449f-a982-8f8c243cf960

Using `watch` with `nextTick` here seems to fix the issue. However, the overlay size is still incorrect when the page is loaded:

https://github.com/samatechtw/pubstudio-builder/assets/59676941/c509d1a6-3477-42b1-9746-127fbc5a00af

I wonder if using an interval to periodically update the overlay style would make things easier; but that sounds like a strange hack 😵!